nestkernel/random_generators.h

```diff
@@ -115,6 +115,13 @@ class BaseRandomGenerator
    * @param N Maximum value that can be drawn.
    */
   virtual unsigned long ulrand( unsigned long N ) = 0;
+
+  /**
+   * @brief Uses the wrapped RNG engine to draw an unsigned long from a Poisson distribution with mean mu.
+   *
+   * @param mu Mean value of Poisson distribution.
+   */
+  virtual unsigned long prand( double mu ) = 0;
```

Comment:
You should not introduce new random number functions here, the interface is limited to `drand()` (for double-valued densities) and `ulrand()` (for integer-valued distributions). You need to create a `poisson_distribution` object in the connection builder and obtain the random number from it, similar to the `FixedTotalNumber` builder:

https://github.com/nest/nest-simulator/blob/9575a63b2c65cab5ea1f2bc092a86ba1acccb95a/nestkernel/conn_builder.cpp#L1385

Reply:
Addressed in 29f6d751013b363b90ed241b363c7205d37cfb60.